### PR TITLE
fix(headers) admin api and proxy server headers adjustment

### LINKED ISSUE #6367: setting config closing the client response failed
### Summary

i wanted to hide the `Response Headers` `Server`, but i found that after I set the `KONG_HEADERS=off` env parameter, i could still see it (i also use the `/etc/kong/kong.conf.default` file but also have no effect)
```
 # Comma-separated list of headers Kong should inject in client responses.
headers = off
```

### Steps To Reproduce
docker-compose
1. 
```yaml
  kong:
    container_name: "kong"
    ports:
      - "8000:8000"
      - "8443:8443"
      - "8001:8001"
      - "8444:8444"
    environment:
      - TZ=Asia/Shanghai
      - KONG_DATABASE=postgres
      - KONG_PG_HOST=kong-database
      - KONG_CASSANDRA_CONTACT_POINTS=kong-database
      - KONG_ADMIN_ACCESS_LOG=/dev/stdout
      - KONG_PROXY_ERROR_LOG=/dev/stderr
      - KONG_ADMIN_ERROR_LOG=/dev/stderr
      - KONG_ADMIN_LISTEN=0.0.0.0:8001, 0.0.0.0:8444 ssl
      - KONG_HEADERS=off
    restart: always
    image: kong
```
2. docker-compose up -d

### Additional Details & Logs
```
10.40.30.106 - - [18/Sep/2020:13:52:00 +0800] "GET / HTTP/1.1" 200 9711 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/85.0.4183.102 Safari/537.36"
```
- Kong version (`$ kong version`):  kong `2.1.3` and `1.3.1`
- Operating system: Centos:7

### PR DESCRIPTION
**Pull request by bungle**
### Summary

This commit contains following things:
1. it removes `Server` header setting from `kong.response.exit`
2. admin api now sets `Server` header according to `conf.headers` on `header_filter`
3. proxy does not add `Server` header in case `Via` header contains the same value and request is short-circuited (this is for `aws-lambda`, `azure-functions` etc. plugins that kinda proxy, but do not use the proxy module, and skip balancer)

### Issues Resolved

Fix #6367

### CODE REVIEW

**Comment by xichengliudui** [comment]
to whom do need to assign pr?

**Comment by bungle** [comment]
> to whom do need to assign pr?

Hi, did you have time to try this PR? Does it fix your issue?

File: spec/02-integration/05-proxy/14-server_tokens_spec.lua [hunk 1]
```diff
@@ -430,6 +470,8 @@ describe("headers [#" .. strategy .. "]", function()
           assert.res_status(200, res)
           admin_client:close()
 
+          ngx.sleep(1)
```
Comment (kikito) [inline spec/02-integration/05-proxy/14-server_tokens_spec.lua:473]: Is this needed, or is it a leftover from dev? If needed, do we have an explanation about why?

Comment (bungle) [reply spec/02-integration/05-proxy/14-server_tokens_spec.lua]: I had `lua-resty-worker-events 2.0.0` on my machine where it was required, but it is not required without it in current master, so I removed it.

**Review by kikito — APPROVED** [review]

**Comment by xichengliudui** [comment]
> > to whom do need to assign pr?
> 
> Hi, @xichengliudui did you have time to try this PR? Does it fix your issue?

ok, let me try it

**Pull request merged**